File: tests/conformance/sandbox-tools/fetch_allowed.rb
tool description: "Fetch an allowed domain",
     input: {},
     permissions: { network: ["localhost"] }

execute do |args, ctx|
  if ZeroMcp::Sandbox.check_network_access(ctx.tool_name, "localhost", ctx.permissions)
    { "status" => "ok", "domain" => "localhost" }
  else
    { "status" => "error" }
  end
end

File: tests/conformance/sandbox-tools/fetch_allowed.php
<?php

return [
    'description' => 'Fetch an allowed domain',
    'input' => [],
    'permissions' => ['network' => ['localhost']],
    'execute' => function ($args, $ctx) {
        if (\ZeroMcp\Sandbox::checkNetworkAccess($ctx->toolName, 'localhost', $ctx->permissions)) {
            return ['status' => 'ok', 'domain' => 'localhost'];
        }
        return ['status' => 'error'];
    },
];
